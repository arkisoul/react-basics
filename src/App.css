.main-content {
    background-color: lavenderblush;
    min-height: 100vh;
}
.btn {
  all: unset;
  padding: 0.75em 1em;
  border-radius: 0.5em 0 0.5em 0;
  font-size: 0.9rem;
  text-align: center;
  cursor: pointer;
  box-shadow: 0 0 4px rgba(0, 0, 0, 0.25);
  transition: all 300ms ease-in;
}
.btn-primary {
  background-color: navajowhite;
}
.btn-accent {
  background-color: royalblue;
  color: white;
}
.btn-block {
  flex: 1 1 100%;
}
.btn:hover {
  transform: scale(1.1);
}
