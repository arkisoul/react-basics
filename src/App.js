import React, { Suspense } from "react";
import {
  BrowserRouter,
  Route,
  Switch,
  Redirect,
  useRouteMatch,
} from "react-router-dom";

import "./App.css";
import { DashboardComponent } from "./views";

function App() {
  const accessToken = localStorage.getItem("accessToken");

  return (
    <BrowserRouter>
      <main className="main-content">
        <Suspense fallback={<div>Loading...</div>}>
          <Switch>
            <Route exact path="/">
              <h1>Login Page</h1>
            </Route>
            <Route exact path="/login">
              <h1>Login Page</h1>
            </Route>
            <Route path="/dashboard">
              {accessToken ? <DashboardComponent /> : <Redirect to="/login" />}
            </Route>
            <Route path="*">
              <div className="NotFoundPage">
                <h1>We could not find what you are looking for.</h1>
              </div>
            </Route>
          </Switch>
        </Suspense>
      </main>
    </BrowserRouter>
  );
}

export default App;
