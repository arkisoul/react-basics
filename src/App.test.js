import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

describe('App Component', () => {
  test('should render the component', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
