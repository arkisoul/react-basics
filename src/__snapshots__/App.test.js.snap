// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render the component 1`] = `
<BrowserRouter>
  <main
    className="main-content"
  >
    <Suspense
      fallback={
        <div>
          Loading...
        </div>
      }
    >
      <Switch>
        <Route
          exact={true}
          path="/"
        >
          <h1>
            Login Page
          </h1>
        </Route>
        <Route
          exact={true}
          path="/login"
        >
          <h1>
            Login Page
          </h1>
        </Route>
        <Route
          path="/dashboard"
        >
          <Redirect
            to="/login"
          />
        </Route>
        <Route
          path="*"
        >
          <div
            className="NotFoundPage"
          >
            <h1>
              We could not find what you are looking for.
            </h1>
          </div>
        </Route>
      </Switch>
    </Suspense>
  </main>
</BrowserRouter>
`;
